numpy/lib/function_base.py

```diff
@@ -764,6 +764,22 @@ def copy(a, order='K', subok=False):
     >>> x[0] == z[0]
     False
 
+    Note that ``np.copy`` is a shallow copy and will not
+    copy object elements within arrays:
```

Comment:
Not my original wording was also trying to cover arrays with dtypes like `np.dtype([('a', int), ('b', object)])` - so `dtype=object` is too specific.

Reply:
I see - I wasn't thinking of structured dtypes that contained `object` when I suggested the above change, but making clear that it is the dtype, not the type of the individual elements, that governs this behavior. I'm probably splitting hairs so feel free to ignore the suggestions!

Reply:
Perhaps @seberg can work out a sensible wording.

Reply:
"shallow" is probably a good word, even if not easy on beginners. Maybe we can follow with something along: `This is (mainly?) important for arrays containing python objects. The new array will contain the same object which may lead to surprises if that object can be modified (mutable)::`
In reality, I suppose it would be true for any "mutable" scalar element, but we do not really have those, except object. And then we have structured which fits neither category, and we should just ignore it :). If we want to be super clear, I guess we could explain why `dict()` and what `b[0] is a[0]` actually proofs with a short comment. But I guess its not our job, and in principle the best we could do i theory is to link to beginner documentation which explains mutability.

Reply:
It sounds like the original wording (pre my suggestion) fits better with the consensus - thanks for weighing in @seberg !

Reply:
Thanks for the suggestions everyone. So, can I update the document with the following?

> Note that ``np.copy`` is a shallow copy and will not copy object elements within arrays. This is mainly important for arrays containing python objects. The new array will contain the same object which may lead to surprises if that object can be modified (mutable):

Reply:
Perhaps "(is mutable)" would read better. I pointing out that `copy.deepcopy` works fine on arrays  might be good to fold in.

Reply:
Thanks for the tip @eric-wieser.

> Pointing out that copy.deepcopy works fine on arrays might be good to fold in.
There's a small paragraph right after with this info.